sdk/ochopod/models/piped.py

```diff
@@ -349,14 +350,16 @@ def on(self, data):
 
                     #
                     # - combine our environment variables with the overrides from configure()
-                    # - popen() the new process
+                    # - popen() the new process and log stdout/stderr in separate thread
                     # - reset the sanity check counter
                     # - keep track of its pid to kill it later on
                     #
                     env = deepcopy(self.env)
                     env.update(data.env)
                     tokens = data.command if self.shell else data.command.split(' ')
-                    data.sub = Popen(tokens, cwd=self.cwd, env=env, shell=self.shell)
+                    data.sub = Popen(tokens, cwd=self.cwd, env=env, shell=self.shell, stdout=PIPE, stderr=STDOUT)
+                    if self.log_proc:
+                        self.start_proc(data)
```

Comment:
just init the logging thread in there, no need to catch (raising will fail the configure() in any case).